File: old/ressources.lua
Ressources = {
    sounds = {},
    tilesetImage,
    tileSize,
    tileQuads = {},
    tilesetBatch
}


function Ressources.load()
    Ressources.sounds["nothing"] = love.audio.newSource("nothing.wav", "static")
    Ressources.sounds["loose"]   = love.audio.newSource("loose.wav"  , "static")
    Ressources.sounds["start"]   = love.audio.newSource("start.wav"  , "static")
    Ressources.sounds["win"]     = love.audio.newSource("win.wav"    , "static")
end 

function Ressources.play(sound)
	love.audio.rewind(Ressources.sounds[sound])
	love.audio.play(Ressources.sounds[sound])
end 



function Ressources.setupTileset()
	Ressources.tilesetImage = love.graphics.newImage( "minesweeper.png" )
	Ressources.tilesetImage:setFilter("nearest", "nearest")
	Ressources.tileSize = 20

	widthInTile  =  Ressources.tilesetImage:getWidth()/Ressources.tileSize;
	heightInTile =  Ressources.tilesetImage:getHeight()/Ressources.tileSize

	for i=0,widthInTile-1 do
		for j=0,heightInTile-1 do 	
			Ressources.tileQuads[i+widthInTile*j] =  love.graphics.newQuad(i * Ressources.tileSize, j * Ressources.tileSize, Ressources.tileSize, Ressources.tileSize, Ressources.tilesetImage:getWidth(), Ressources.tilesetImage:getHeight())
		end
	end
end

function Ressources.initTilesetBatch(mapWidth,mapHeight,map,tileFromModel)
	Ressources.tilesetBatch = love.graphics.newSpriteBatch(Ressources.tilesetImage, mapWidth * mapHeight)
	Ressources.updateTilesetBatch(mapWidth,mapHeight,map,tileFromModel)

end

function Ressources.updateTilesetBatch(tilesDisplayWidth,tilesDisplayHeight,map,tileFromModel)
	Ressources.tilesetBatch:clear()
	
	for x=1, tilesDisplayWidth do
		for y=1, tilesDisplayHeight do
			local id = tileFromModel(map[x][y])
			Ressources.tilesetBatch:addq(Ressources.tileQuads[id], (x-1)*Ressources.tileSize, (y-1)*Ressources.tileSize)
		end
	end
end

function Ressources.paint(zoomX,zoomY)
    mapX=1
    mapY=1
	love.graphics.draw(Ressources.tilesetBatch,0,0,0, zoomX, zoomY)

end

File: src/ressources.ts
export type SoundType = "nothing" | "loose" | "start" | "win" 

export class Ressource{
	sounds:Record<SoundType,Source>;
	image:Image;
	quads:Array<Quad>;
	tileSize:number;
	batch:SpriteBatch;

	constructor() {
		this.sounds = {
			nothing: love.audio.newSource("assets/nothing.wav", "static"),
			loose: love.audio.newSource("assets/loose.wav", "static"),
			start: love.audio.newSource("assets/start.wav", "static"),
			win: love.audio.newSource("assets/win.wav", "static"),
		};
		this.image = love.graphics.newImage( "assets/minesweeper.png" );
		this.image.setFilter("nearest", "nearest")
		this.quads = [];
		this.tileSize=20;

		let widthInTile  : number = this.image.getWidth()/this.tileSize;
		let heightInTile : number = this.image.getHeight()/this.tileSize;

		for(let i=0;i<widthInTile;i++){
			for(let j=0;j<heightInTile;j++){
				this.quads[i+widthInTile*j] =  love.graphics.newQuad(
					i * this.tileSize,
					j * this.tileSize,
					this.tileSize,
					this.tileSize,
					this.image.getWidth(),
					this.image.getHeight(),
				)      
			}    
		}
	}

	initBatch(w :number,h:number){
		this.batch = love.graphics.newSpriteBatch(this.image, w*h)
	}

	updateTilesetBatch(w :number,h:number,tiles:Array<Array<number>>){
		this.batch.clear()
		for(let i=0;i<w;i++){
			for(let j=0;j<h;j++){
				let id = tiles[i][j]
				this.batch.add(
					this.quads[id],
					 (i-1)*this.tileSize, 
					 (j-1)*this.tileSize
				)
			}
		}
	}

	play(st: SoundType){
		love.audio.play(this.sounds[st])
	}
}


